src/LibraryManager.Vsix/Json/Completion/LibraryIdCompletionProvider.cs

```diff
@@ -77,17 +76,26 @@ protected override IEnumerable<JSONCompletionEntry> GetEntries(JSONCompletionCon
 
                 if (set.Completions != null)
                 {
+                    bool versionSpecific = false;
+                    bool setVersionSpecific = true;
+
                     foreach (CompletionItem item in set.Completions)
                     {
+                        if (setVersionSpecific)
+                        {
+                            versionSpecific = item.DisplayText.Contains("@");
+                            setVersionSpecific = false;
+                        }
+
                         string insertionText = item.InsertionText.Replace("\\\\", "\\").Replace("\\", "\\\\");
```

Comment:
>string insertionText = item.InsertionText.Replace("\\\\", "\\").Replace("\\", "\\\\"); [](start = 24, length = 86)

This is confusing to me. Can you comment why this is happening?

Reply:
I don't understand this line. It also confuses me...

---
In reply to: [198552566](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198552566) [](ancestors = 198552566)

Reply:
Can you add a comment:
"Make sure to double-escape any singly-escaped backslashes"

---
In reply to: [198653332](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198653332) [](ancestors = 198653332,198552566)

Reply:
Sure will do

---
In reply to: [199295753](https://github.com/aspnet/LibraryManager/pull/258#discussion_r199295753) [](ancestors = 199295753,198653332,198552566)

Reply:
This is marked as resolved, but it hasn't been explained yet. Can you please clarify in the code why this is happening?

---
In reply to: [199372324](https://github.com/aspnet/LibraryManager/pull/258#discussion_r199372324) [](ancestors = 199372324,199295753,198653332,198552566)